openfe/protocols/openmm_utils/omm_compute.py

```diff
@@ -1,23 +1,39 @@
 # This code is part of OpenFE and is licensed under the MIT license.
 # For details, see https://github.com/OpenFreeEnergy/openfe
 # Adapted Perses' perses.app.setup_relative_calculation.get_openmm_platform
+from typing import Optional
 import warnings
 import logging
 
 
 logger = logging.getLogger(__name__)
 
 
-def get_openmm_platform(platform_name=None):
+def get_openmm_platform(
+    platform_name: Optional[str] = None,
+    gpu_device_index: Optional[list[int]] = None,
+    restrict_cpu_count: bool = False
+):
     """
     Return OpenMM's platform object based on given name. Setting to mixed
     precision if using CUDA or OpenCL.
 
     Parameters
     ----------
-    platform_name : str, optional, default=None
+    platform_name : Optional[str]
         String with the platform name. If None, it will use the fastest
         platform supporting mixed precision.
+        Default ``None``.
+    gpu_device_index : Optional[list[str]]
```

Comment:
Actually we should probably have a chat about how we handle this long term - this is a bit like MPI settings, where technically we shouldn't make this immutable but maybe something we pick up at run time?

How can we go about handling this properly?

Reply:
Something I don't think we abstracted well are "run time arguments". Like we have the split for settings that change thermo, but didn't consider a category of non-thermo settings that make the most sense to pick at runtime, I haven't looked at the code yet and will update this comment, but I suspect what we should do is 

1) have some default
2) read this setting
3) read in an environmental variable  

If we do things in that order, it means we don't break anything old, then when configuring your system you can make some choices, but then when running on HPC you can still set things if needed and override the settings